text/0095-nft-standard.md

```diff
@@ -0,0 +1,364 @@
+            }
+            None => None,
+        };
+        runtime::ret(CLValue::from_t(symbol).unwrap_or_revert());
+    } 
+    ```
+
+## Token Metadata Extension
+
+[token-metadata-extension]: #token-metadata-extension
+
+### Hybrid Approach for Token Metadata in CEP-95
+
+[hybrid-approach-for-token-metadata-in-cep-95]: #hybrid-approach-for-token-metadata-in-cep-95
+
+The CEP-95 standard provides a flexible, hybrid approach for NFT token metadata, allowing developers and consumers to leverage **both on-chain and off-chain metadata sources**.
+
+- **On-chain metadata** (`token_metadata`):
+  - This method allows direct storage of metadata within the blockchain. Metadata is stored as key-value pairs (`Vec<(String, String)>`), giving a clear yet flexible structure without enforcing a specific serialization format. When metadata is created or updated, the contract emits a `MetadataUpdate` event, ensuring that indexers and clients are immediately informed and can cache or display this data without additional lookups. 
+- **Off-chain metadata** (`token_uri`):
+  - Additionally, CEP-95 supports traditional off-chain metadata retrieval via a URI. Contracts can provide a URI linking to external JSON metadata (e.g., hosted on IPFS or HTTP servers). This off-chain approach reduces on-chain storage costs and allows easier updates of non-critical metadata fields.
+
+Consumers, such as wallets, explorers, and marketplaces, benefit from this hybrid approach by first attempting to retrieve on-chain metadata for immediate trustworthiness and availability. If on-chain metadata is unavailable or limited, clients can fall back on the provided metadata URI. This method offers the best balance between efficiency, flexibility, trust, and ease of use, ensuring a smooth and user-friendly NFT experience within the Casper ecosystem.
+
+### Token Metadata Interface: CEP95MetadataEvent
+
+[token-metadata-interface-cep95metadataevent]: #token-metadata-interface-cep95metadataevent
+
+```rust
+pub enum CEP95MetadataEvent {
+    /// Emitted whenever on-chain metadata for a token is created or updated.
+    ///
+    /// @param token_id - The ID of the NFT whose metadata was updated
+    /// @param metadata - The raw metadata (e.g., JSON bytes)
+    MetadataUpdate {
+        token_id: U256,
+    },
+}
+
+/// Optional interface for retrieving token metadata
+pub trait CEP95TokenMetadata {
+    /// Returns metadata for a given token ID.
+    ///
+    /// @param token_id - The NFT ID.
+    /// @return Option<Vec<(String, String)>> - optional metadata stored as a list of key-value pairs.
+    fn token_metadata(&self, token_id: U256) -> Option<Vec<(String, String)>>;
+    
+    /// Returns a URI pointing to metadata for a specific token.
+    ///
+    /// @param token_id - The NFT ID.
+    /// @return Option<String> - A URI pointing to metadata if available.
+    fn token_uri(&self, token_id: U256) -> Option<String>;
+}
+```
+
+### CEP-95 On-chain Metadata Schema
```

Comment:
I'd name it "On-chain token metadata" and would add a separate "Off-chain token metadata"